iina/Logger.swift

```diff
@@ -154,8 +155,15 @@ struct Logger {
     guard enabled else { return }
     #endif
 
-    guard level >= .preferred else { return }
-    let string = formatMessage(message, level, subsystem, appendNewlineAtTheEnd)
+    guard level.rawValue >= Preference.integer(for: .logLevel) else { return }
+
+    let date = Date()
+    let string = formatMessage(message, level, subsystem, appendNewlineAtTheEnd, date)
+    let log = Log(subsystem: subsystem.rawValue, level: level.rawValue, message: message, date: dateFormatter.string(from: date), logString: string)
+    DispatchQueue.main.async {
+      (NSApp.delegate as! AppDelegate).logWindow.logs.append(log)
+    }
+
```

Comment:
1. Did you measure the performance impact of reading the preference each time? We can add observer to `UserDefaults` and update `Log.Level.preferred` accordingly (like what we're doing for lots of other preference keys).
2. Please make log window `lazy` and save the logs elsewhere. There's no need to force initializing the log window (and even write the logs in main thread). I'd prefer moving the subsystems to directly under `Logger`, too:
  ```swift
  struct Logger {
    class Log: NSObject {}
    static var logs: [Log]
    static var subsystems: [Subsystems]
  }
  ```